service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java

```diff
@@ -10,4 +19,58 @@
   public SiteProcessorImpl(DSLContext db) {
     this.db = db;
   }
+
+  private void checkSiteExists(int siteId) {
+    if (!db.fetchExists(db.selectFrom(SITES).where(SITES.ID.eq(siteId)))) {
+      throw new ResourceDoesNotExistException(siteId, "Site");
+    }
+  }
+
+  private Boolean isAlreadyFavorite(int userId, int siteId) {
+    return db.fetchExists(
+        db.selectFrom(FAVORITE_SITES)
+            .where(FAVORITE_SITES.USER_ID.eq(userId))
+            .and(FAVORITE_SITES.SITE_ID.eq(siteId)));
+  }
+
+  @Override
+  public void favoriteSite(JWTData userData, int siteId) {
+    checkSiteExists(siteId);
+    if (isAlreadyFavorite(userData.getUserId(), siteId)) {
+      // throw exception
+    }
+
+    FavoriteSitesRecord record = db.newRecord(FAVORITE_SITES);
+    record.setUserId(userData.getUserId());
+    record.setSiteId(siteId);
+    record.store();
+  }
+
+  @Override
+  public void unfavoriteSite(JWTData userData, int siteId) {
+    checkSiteExists(siteId);
+    if (isAlreadyFavorite(userData.getUserId(), siteId)) {
+      throw new ResourceDoesNotExistException(-1, "FavoriteSiteRecord");
```

Comment:
Could replace this with a WrongFavoriteStatus type exception that can be used in the favorite site function too

Reply:
Done